Tests/NIOTests/CodecTest.swift

```diff
@@ -377,4 +377,76 @@ public class MessageToByteEncoderTest: XCTestCase {
         XCTAssertFalse(try channel.finish())
 
     }
+    
+    struct fixedLengthFrameDecoderHelper<T: FixedWidthInteger> {
```

Comment:
I think I would move the tests to an extra file as the codec itself is also in an extra file.